usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawZoneClient.js

```diff
@@ -0,0 +1,725 @@
+        { red: 241, green: 243, blue: 238 }, // Neutral 1
+        { red: 23, green: 41, blue: 131 }, // Blue
+        { red: 0, green: 158, blue: 224 }, // Cyan
+        { red: 0, green: 144, blue: 54 }, // Green
+        { red: 255, green: 237, blue: 0 } // Yellow
+    ];
+    var HIFI_COLORS_URLS = [
+        Script.resolvePath("../resources/textures/black.png"),
+        Script.resolvePath("../resources/textures/red.png"),
+        Script.resolvePath("../resources/textures/magenta.png"),
+        Script.resolvePath("../resources/textures/neutral4.png"),
+        Script.resolvePath("../resources/textures/neutral3.png"),
+        Script.resolvePath("../resources/textures/neutral2.png"),
+        Script.resolvePath("../resources/textures/neutral1.png"),
+        Script.resolvePath("../resources/textures/blue.png"),
+        Script.resolvePath("../resources/textures/cyan.png"),
+        Script.resolvePath("../resources/textures/green.png"),
+        Script.resolvePath("../resources/textures/yellow.png")
+    ];
+    var MINIMUM_TRIGGER_PRESS_VALUE = 0.97;
+    var REPEAT_DISTANCE_CHECK_MS = 60;
+    var MINIMUM_MOVEMENT_TO_DRAW_M = 0.0005;
+    var DEFAULT_NORMAL = { x: 0, y: 0, z: 1 };
+    var DECAY_TIME_S = 60;
+    var MAX_LINE_POINTS = 100;
+    var DRAW_SOUND = SoundCache.getSound(Script.resolvePath('../resources/sounds/draw.mp3'));
+    var DRAW_SOUND_VOLUME = 0.01;
+    var DELETE_AGAIN_MS = 100;
+    var MAXIMUM_DISTANCE_TO_SEARCH_M = 1;
+    var MAXIMUM_DISTANCE_TO_DELETE_M = 0.03;
+    var DISTANCE_TO_DRAW_IN_FRONT_OF_CAMERA_DESKTOP_M = 1.5;
+    var OPEN_SOUND = SoundCache.getSound(Script.resolvePath('../resources/sounds/open.mp3'));
+    var OPEN_SOUND_VOLUME = 0.2;
+    var CLOSE_SOUND = SoundCache.getSound(Script.resolvePath('../resources/sounds/close.mp3'));
+    var CLOSE_SOUND_VOLUME = 0.3;
+    var WAIT_TO_CLEAN_UP_MS = 2000;
+    var WAIT_TO_REOPEN_APP_MS = 500;
+
+
+    var scannerZoneMarker;
+    var injector;
+    var paintSphere;
+    var paintSphereMaterial;
+    var parentJointIndex;
+    var randomHiFiColorIndex;
+    var controllerMapping;
+    var controllerMappingName = 'Hifi-DrawApp';
+    var activeTriggerPress = false;
+    var activeGripPress = false;
+    var distanceCheckInterval = null;
+    var polyLine = null;
+    var lineStartPosition;
+    var previousLinePoint;
+    var linePoints = [{x: 0, y: 0, z: 0 }];
+    var lineNormals = [DEFAULT_NORMAL, DEFAULT_NORMAL];
+    var lineStrokeWidths = [];
+    var paintSphereDimensions;
+    var deletingInterval;
+    var previousLinePointDesktop;
+    var pickRay;
+    var desktopActionProgress = false;
+    var animationData = {};
+    var animationHandlerID;
+    var mouseEventsConnected = false;
+    var dominantHandJoint;
+    var dominantHand;
+    var tablet = Tablet.getTablet('com.highfidelity.interface.tablet.system');
+    var button;
+
+    var DrawingZone = function() {
+        _this = this;
+    };
+
+    DrawingZone.prototype = {
```

Comment:
much of the logic in here seems to be duplicated from `drawSphereClient.js` - is that accurate?